src/Features/LanguageServer/ProtocolUnitTests/Diagnostics/AbstractPullDiagnosticTestsBase.cs

```diff
@@ -158,29 +160,30 @@ private protected override TestAnalyzerReferenceByLanguage CreateTestAnalyzersRe
             return new WorkspaceDiagnosticParams
             {
                 PreviousResultId = previousResultsLsp,
-                PartialResultToken = progress
+                PartialResultToken = progress,
+                Identifier = category
             };
         }
 
         private static TestDiagnosticResult ConvertWorkspaceDiagnosticResult(SumType<WorkspaceFullDocumentDiagnosticReport, WorkspaceUnchangedDocumentDiagnosticReport> workspaceReport)
         {
             if (workspaceReport.Value is WorkspaceFullDocumentDiagnosticReport fullReport)
             {
-                return new TestDiagnosticResult(new TextDocumentIdentifier { Uri = fullReport.Uri }, fullReport.ResultId!, fullReport.Items);
+                return new TestDiagnosticResult(new TextDocumentIdentifier { Uri = fullReport.Uri }, fullReport.ResultId, fullReport.Items);
             }
             else
             {
                 var unchangedReport = (WorkspaceUnchangedDocumentDiagnosticReport)workspaceReport.Value!;
-                return new TestDiagnosticResult(new TextDocumentIdentifier { Uri = unchangedReport.Uri }, unchangedReport.ResultId!, null);
+                return new TestDiagnosticResult(new TextDocumentIdentifier { Uri = unchangedReport.Uri }, unchangedReport.ResultId, null);
             }
         }
 
         private protected static Task CloseDocumentAsync(TestLspServer testLspServer, Document document) => testLspServer.CloseDocumentAsync(document.GetURI());
 
         private protected static ImmutableArray<(string resultId, TextDocumentIdentifier identifier)> CreateDiagnosticParamsFromPreviousReports(ImmutableArray<TestDiagnosticResult> results)
         {
-
-            return results.Select(r => (r.ResultId, r.TextDocument)).ToImmutableArray();
+            // If there was no resultId provided in the response, we cannot create previous results for it.
+            return results.Where(r => r.ResultId != null).Select(r => (r.ResultId!, r.TextDocument)).ToImmutableArray();
```

Comment:
fixes a bug in the tests discovered when enabling categories for public workspace diagnoistics - we incorrectly sent back a non-null `PreviousResultId` object with a null `resultId` (not allowed)